aten/src/ATen/native/transformers/cuda/attention_backward.cu

```diff
@@ -76,7 +76,7 @@ std::tuple<Tensor, Tensor, Tensor> _flash_attention_backward(
 
   // We check the whether the cumulative_sequence_length_q is defined
   // in order to determine whether we are using varlen or dense forward
-  if (cumulative_sequence_length_q.defined()) {
+  if (cumulative_sequence_length_q.defined() && cumulative_sequence_length_q.numel() > 0) {
```

Comment:
I don't love this, why do we need both?

Reply:
I was being exhaustive, as C++ calls from SDPA for dense will use Tensor(), while from python I am passing a torch.empty(0) tensor. This should cover both cases? In any case, probably numel only will cover both cases

Reply:
I think if we revert the other changes though we would just pass in cum_sum = None from python right?

Reply:
yes, I'll do it now